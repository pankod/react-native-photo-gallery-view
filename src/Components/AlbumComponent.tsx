// Global Imports
import React, { Component } from 'react';
import { Text, View, Image, TouchableOpacity, Dimensions, ScrollView, FlatList } from 'react-native';

// Local Imports
import { AlbumStyle } from "@Styles";
import { IAlbumProps, IMediaItem, IGalleryProps, IGalleryState } from "@Interfaces";
import Common from '@Provider';

const { width, height } = Dimensions.get('window');

export class AlbumComponent extends Component<IAlbumProps, {}> {

	static contextType = Common

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<FlatList style={AlbumStyle.container} data={context.mediaList}
							numColumns={context.gridSize}
							renderItem={({ item, index }) => this.renderItem(item, index)}
							keyExtractor={(item, index) => index.toString()}
						/>
					)
				}
			</Common.Consumer>
		)
	}

	private renderItem(item: IMediaItem, index: number): JSX.Element {
		const { gridSize, onSelectionChanged } = this.context;
		return (
			<TouchableOpacity onPress={() => onSelectionChanged(item, index)} key={index} style={{ width: width / gridSize, height: width / gridSize }}>
				<Image resizeMode={"cover"} style={{ flex: 1 }} key={index} source={{ uri: item.thumb }} />
			</TouchableOpacity>
		)
	}
}
