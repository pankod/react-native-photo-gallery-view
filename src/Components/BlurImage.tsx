// Global Imports
import React, { PureComponent } from 'react'
import { Animated } from 'react-native'

// Local Imports
import { IBlurImageProps } from '@Interfaces';
import { BlurImageStyle } from '@Styles';

export class BlurImage extends PureComponent<IBlurImageProps> {

	public imageAnimated = new Animated.Value(0);

	private onImageLoad(): void {
		Animated.timing(this.imageAnimated, {
			toValue: 1,
			duration: 500,
			useNativeDriver: true
		}).start();
	}

	render() {
		return (
			<React.Fragment>
				<Animated.Image
					{...this.props}
					style={[BlurImageStyle.image, { opacity: this.imageAnimated }]}
					onLoad={this.onImageLoad.bind(this)}
					blurRadius={1}
				/>
			</React.Fragment>
		)
	}
}
