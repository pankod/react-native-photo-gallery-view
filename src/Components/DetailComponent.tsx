// Global Imports
import React, { PureComponent } from 'react'

// Local Imports
import Common from '@Provider';
import { IGalleryProps, IGalleryState } from '@Interfaces';
import { BlurImage } from '@Components';

export class DetailComponent extends PureComponent {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<BlurImage
							resizeMethod={"resize"}
							resizeMode={"contain"}
							source={{ uri: context.showingImage && context.showingImage.photo }}
							thumbnail={{ uri: context.showingImage && context.showingImage.thumb }}
						/>
					)
				}
			</Common.Consumer>
		)
	}
}
