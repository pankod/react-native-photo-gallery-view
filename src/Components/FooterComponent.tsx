// Global Imports
import React, { PureComponent } from 'react';
import { View } from 'react-native';

// Local Imports
import { FooterStyle } from "@Styles";
import { IFooterProps, IGalleryProps, IGalleryState } from "@Interfaces";
import Common from '@Provider';
import { Const } from '@Constants';

export class FooterComponent extends PureComponent<IFooterProps, {}> {

	static contextType = Common;

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{
					(context: IGalleryProps & IGalleryState) => (
						<View style={[FooterStyle.container, context.customFooterStyle]}>
							{context.renderStickyFooter && this.renderStickyFooter()}
						</View>
					)
				}
			</Common.Consumer>
		)
	}

	private renderStickyFooter(): JSX.Element {
		return (
			this.context.renderStickyFooter(Const.HEIGHT)
		)
	}
}
