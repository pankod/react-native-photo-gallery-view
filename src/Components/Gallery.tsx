import React, { Component } from "react"
import { Text, View } from "react-native"

export class Gallery extends Component {
	render() {
		return (
			<View>
				<Text> textInComponent </Text>
			</View>
		)
	}
}
