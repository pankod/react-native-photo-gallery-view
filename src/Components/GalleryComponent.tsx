// Global Imports
import React, { Component } from "react"
import { View } from "react-native"

// Local Imports
import { TopBarComponent, AlbumComponent, FooterComponent } from "@Components";
import { IGalleryProps, IGalleryState, IMediaItem } from "@Interfaces";
import { GalleryStyle } from "@Styles";
import { Const } from '@Constants';
import Common from '@Provider';

export class GalleryComponent extends Component<IGalleryProps, IGalleryState> {

	constructor(props: IGalleryProps) {
		super(props);
		this.state = {
			title: `${props.mediaList.length} Photos`
		}
	}

	static defaultProps = {
		stickyFooter: true,
		gridSize: 3,
	}

	public render(): JSX.Element {
		const {
			style,
			stickyFooter
		} = this.props;
		return (
			<Common.Provider value={{ ...this.state, ...this.props }}>
				<View style={[GalleryStyle.container, style]}>
					<TopBarComponent />
					<AlbumComponent />
					{stickyFooter &&
						<FooterComponent />
					}
				</View>
			</Common.Provider>
		)
	}
}
