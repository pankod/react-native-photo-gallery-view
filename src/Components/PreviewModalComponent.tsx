// Global Imports
import React, { PureComponent } from 'react'
import { Text, View, Modal, Image, Animated, ImageBackground, Platform } from 'react-native'

// Local Imports
import Common from '@Provider';
import { IGalleryProps, IGalleryState } from '@Interfaces';
import { PreviewModalStyle } from '@Styles';

export class PreviewModal extends PureComponent {

	static contextType = Common;

	public scale = new Animated.Value(0);
	public opacity = new Animated.Value(0);

	public componentDidMount(): void {
		Animated.parallel([
			Animated.spring(this.scale, {
				toValue: 1,
				bounciness: 10,
				useNativeDriver: true
			}),
			Animated.timing(this.opacity, {
				toValue: 1,
				duration: 500,
				useNativeDriver: true
			})
		]).start();
	}

	public render(): JSX.Element {
		return (
			<Common.Consumer>
				{(context: IGalleryProps & IGalleryState) => (
					<Modal
						animationType={"none"}
						visible={context.previewIsOpen}
						transparent={true}
					>
						<ImageBackground style={{
							flex: 1,
						}}
							blurRadius={Platform.OS == 'ios' ? 20 : 15}
							// source={{ uri: context.imagePreview.thumb }}
							source={require("../Assets/Images/radius-bg.png")}
							resizeMode={"cover"}
						>
							<Animated.View
								style={[
									PreviewModalStyle.container,
									{
										transform: [
											{
												scale: this.scale.interpolate({
													inputRange: [0, 1],
													outputRange: [0, 1]
												})
											}
										],
										opacity: this.opacity
									}
								]}>
								<View style={PreviewModalStyle.imagePreview}>
									{
										context.customPreviewComponent ?
											context.customPreviewComponent(context.imagePreview) :
											(
												<Image
													style={{ width: "100%", flex: 1 }}
													source={{ uri: context.imagePreview.photo }}
													resizeMode={"contain"}
												/>
											)
									}
								</View>
							</Animated.View>
						</ImageBackground>
					</Modal>
				)}
			</Common.Consumer>
		)
	}
}
