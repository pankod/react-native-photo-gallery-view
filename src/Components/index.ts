export { GalleryComponent } from '@Components/GalleryComponent';
export { TopBarComponent } from '@Components/TopBarComponent';
export { FooterComponent } from '@Components/FooterComponent';
export { AlbumComponent } from '@Components/AlbumComponent';
export { BlurImage } from '@Components/BlurImage';
export { DetailComponent } from '@Components/DetailComponent';
export { PreviewModalComponent } from '@Components/PreviewModalComponent';
