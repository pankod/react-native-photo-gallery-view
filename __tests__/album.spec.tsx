
import * as React from 'react';

import renderer from "react-test-renderer";
import { ShallowWrapper, shallow, mount } from "enzyme";
import PropTypes from "prop-types";

import Common from "../src/Provider";
import { AlbumComponent } from "../src/Components/AlbumComponent";


const data = [{ "caption": "Quaresma", "id": "db1aa2ad-48a2-4882-b6e9-824c0224315f", "photo": "https://xx.jpg", "state": "Deleted", "thumb": "https://xx.jpg" },
{ "caption": "Quaresma 1", "id": "db1aa2ad-48a2-4882-b6e9-824c0224315d", "photo": "https://xx.jpg", "state": "Deleted", "thumb": "https://xx.jpg" }];

describe("Album Component", () => {
	let wrapper: ShallowWrapper;
	let mounting;
	const state = {
		selected: [],
		onSelection: jest.fn(),
		customCheckedView: undefined,
		dynamicSize: {
			width: 1,
			height: 1
		}
	}
	const props = {
		renderCustomState: jest.fn(),
		customThumbnailImage: jest.fn(),
		showImageModal: jest.fn(),
		onSelection: jest.fn(),
		displaySelectionButtons: true,
		stickyFooter: true
	}
	state.selected.push(data[0].id);

	AlbumComponent.contextType = Common;

	let component = (
		<Common.Provider value={{
			items: [], gridSize: 3, ...state, ...props
		}}>
			<AlbumComponent />
		</Common.Provider>
	);

	beforeEach(() => {
		wrapper = shallow(component).dive({ context: { ...state, ...props } });
		mounting = mount(component, {
			context: { ...state, ...props }
		});
	});

	test('should Album Component is render correctly', () => {
		const render = renderer.create(component).toJSON();

		expect(render).toBeDefined();
		expect(render).toMatchSnapshot();
	});

	test('should render item return correctly onSelection jsx element', () => {
		const spy = spyOn(AlbumComponent.prototype, "renderItem").and.callThrough();
		const instance = mounting.instance() as any;

		instance.renderItem(data[0], 0);

		expect(instance.renderItem(data[0], 0).key).toBe("onSelection");
		instance.renderItem(data[0], 0).props.onPress();
		expect(spy).toHaveBeenCalled();
	});

	test('should render item return correctly showImageModal jsx element', () => {
		const spy = spyOn(AlbumComponent.prototype, "renderItem").and.callThrough();
		const instance = mounting.instance() as any;
		instance.context.displaySelectionButtons = false;
		instance.context.stickyFooter = false;
		instance.context.showPreview = jest.fn(() => true);

		wrapper.setProps({
			renderCustomState: jest.fn(() => true)
		})

		instance.renderItem(data[0], 0);

		expect(instance.renderItem(data[0], 0).key).toBe("showImageModal");
		instance.renderItem(data[0], 0).props.onPress();
		instance.renderItem(data[0], 0).props.onLongPress();
		expect(spy).toHaveBeenCalled();
	});

	test('should checkedctrl function return if item is in array', () => {
		const spy = spyOn(AlbumComponent.prototype, "checkedCtrl").and.callThrough();
		const instance = mounting.instance() as any;

		instance.checkedCtrl(data[0]);

		expect(instance.checkedCtrl(data[0])).toBeTruthy();

		expect(typeof instance.checkedCtrl(data[0])).toEqual("boolean");
		expect(spy).toHaveBeenCalled();
	});

	test("should isChecked function return component when item is checked", () => {
		const spy = spyOn(AlbumComponent.prototype, "isChecked").and.callThrough();
		const instance = mounting.instance() as any;

		instance.isChecked(data[0]);

		expect(instance.isChecked(data[0])).toBeDefined();

		expect(typeof instance.isChecked(data[0])).toBe("object")

		expect(spy).toHaveBeenCalled();
	});

	test("should isChecked function return component when item is not checked", () => {
		const spy = spyOn(AlbumComponent.prototype, "isChecked").and.callThrough();
		const instance = mounting.instance() as any;

		instance.isChecked(data[1]);
		expect(instance.isChecked(data[1])).toBeNull();
		expect(spy).toHaveBeenCalled();
	});

	test("should isChecked function return component when item is checked and customCheckedView enable", () => {
		const spy = spyOn(AlbumComponent.prototype, "isChecked").and.callThrough();
		const instance = mounting.instance() as any;
		instance.context.customCheckedView = jest.fn(() => true);

		instance.isChecked(data[0]);

		expect(instance.isChecked(data[0])).toBeTruthy();

		expect(spy).toHaveBeenCalled();
	});

});
