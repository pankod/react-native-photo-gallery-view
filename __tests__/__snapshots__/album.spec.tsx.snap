// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Component should Album Component is render correctly 1`] = `<react-native-mock />`;
